TodoApp/test/TodoApp.Services.Tests/Services/RetrievedItemTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using NSubstitute;
+using NUnit.Framework;
+using TodoApp.Contracts.Models;
+using TodoApp.Contracts.Repositories;
+using TodoApp.Contracts.Services;
+using TodoApp.Services.Services;
+
+namespace TodoApp.Services.Tests.Services
+{
+    [TestFixture]
+    internal class RetrievedItemTests
+    {
+        private IItemRepository _repository;
+        private IGetItemByIdService _getItemByIdService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _repository = Substitute.For<IItemRepository>();
+
+            _getItemByIdService = new GetItemByIdService(_repository);
+        }
+
+        [Test]
+        public async Task GetItemByIdAsync_ExistingId_ReturnsRetrievedItemWithSuccess()
+        {
+            var time = new DateTime(2018, 5, 4, 3, 2, 1);
+            var id = new Guid("44e506f7-c42b-42e5-823c-a5f1cf2e16f9");
+            var item = new Item
+            {
+                Id = id,
+                Text = "testText",
+                CreatedAt = time,
+                ModifiedAt = time
+            };
+            _repository.GetByIdAsync(id).Returns(item);
+
+            var testItem = await _getItemByIdService.GetItemByIdAsync(id);
+
+            Assert.That(testItem.WasFound, Is.True);
+        }
+
+        [Test]
+        public async Task GetItemByIdAsync_NonExistingId_ReturnsItemWithFalseStatus()
```

Comment:
Aha, we didn't understand each other. This test belongs to the `GetItemByIdServiceTests` class.